storage/src/test/java/org/apache/kafka/tiered/storage/utils/TieredStorageTestUtils.java

```diff
@@ -0,0 +1,183 @@
+        overridingProps.setProperty(KafkaConfig.LogCleanupIntervalMsProp(), LOG_CLEANUP_INTERVAL_MS.toString());
+        // The directory of the second-tier storage needs to be constant across all instances of storage managers
+        // in every broker and throughout the test. Indeed, as brokers are restarted during the test.
+        // You can override this property with a fixed path of your choice if you wish to use a non-temporary
+        // directory to access its content after a test terminated.
+        overridingProps.setProperty(storageConfigPrefix(testClassName, STORAGE_DIR_CONFIG), storageDirPath);
+        // This configuration will remove all the remote files when close is called in remote storage manager.
+        // Storage manager close is being called while the server is actively processing the socket requests,
+        // so enabling this config can break the existing tests.
+        // NOTE: When using TestUtils#tempDir(), the folder gets deleted when VM terminates.
+        overridingProps.setProperty(storageConfigPrefix(testClassName, DELETE_ON_CLOSE_CONFIG), String.valueOf(deleteOnClose));
+        return overridingProps;
+    }
+
+    public static Map<String, String> createTopicConfigForRemoteStorage(boolean enableRemoteStorage,
+            long localLogRetentionBytes,
+            int maxRecordBatchPerSegment) {
+        Map<String, String> topicProps = new HashMap<>();
+        // Enables remote log storage for this topic.
+        topicProps.put(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, String.valueOf(enableRemoteStorage));
+        // Ensure offset and time indexes are generated for every record.
+        topicProps.put(TopicConfig.INDEX_INTERVAL_BYTES_CONFIG, "1");
+        // Leverage the use of the segment index size to create a log-segment accepting one and only one record.
+        // The minimum size of the indexes is that of an entry, which is 8 for the offset index and 12 for the
+        // time index. Hence, since the topic is configured to generate index entries for every record with, for
+        // a "small" number of records (i.e. such that the average record size times the number of records is
+        // much less than the segment size), the number of records which hold in a segment is the multiple of 12
+        // defined below.
+        topicProps.put(TopicConfig.SEGMENT_INDEX_BYTES_CONFIG, String.valueOf(12 * maxRecordBatchPerSegment));
+        // To verify records physically absent from Kafka's storage can be consumed via the second tier storage, we
+        // want to delete log segments as soon as possible. When tiered storage is active, an inactive log
+        // segment is not eligible for deletion until it has been offloaded, which guarantees all segments
+        // should be offloaded before deletion, and their consumption is possible thereafter.
+        topicProps.put(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, String.valueOf(localLogRetentionBytes));
```

Comment:
It's confusing we have `localLogRetentionBytes` as a parameter, and put the comments here at the same time. I notice the value of `localLogRetentionBytes` is always 1 in all cases, could we remove it as a parameter? If we expect the parameter will be used in other test cases, then we should move these comment to the places set `localLogRetentionBytes=1`.

Reply:
removed the `localLogRetentionBytes` as parameter. We can extract the parameter later when required.